File: tp1/source/best_first_search.py
from heuristic_search import HeuristicSearch


class BestFirstSearch(HeuristicSearch):

    def _update_frontier(self, child):
        if not (child.state in self.explored or child.state in self.frontier):
            self.frontier.insert((self._heuristic_value(child), child))

    def _heuristic_value(self, node):
        return self.heuristic(node.state, self.problem.goal)

File: tp1/source/best_first_search.cpp

void BestFirstSearch::update_frontier(const Node& child, const ExploredSet& explored, Frontier& frontier){
    if (!(explored.contains(child.state) || frontier.contains(child.state))){
        frontier.insert(child);
    }
}

double BestFirstSearch::heuristic_value(const Node& node){
    return this.heuristic(node.state, this.problem.goal);
}

